lib/io/stream/writable-object.ts

```diff
@@ -0,0 +1,7 @@
+/**
+ * Interface for core functionality of write streams.
+ */
+export interface WritableObject {
```

Comment:
I would choose a different name WritableObject is a little bit ambiguous.  From the functionality it seems that it allow the user of the interface to write string. So can be StringWriter or StringSink ..

Reply:
I have decided on `Writable` because of only one function `write`. I hope it is ok. If not, I will fix it.